projects/plugins/jetpack/extensions/blocks/author-recommendation/edit.js

```diff
@@ -1,26 +1,121 @@
-import { BlockIcon } from '@wordpress/block-editor';
-import { Placeholder } from '@wordpress/components';
+import { BlockIcon, useBlockProps } from '@wordpress/block-editor';
+import {
+	CheckboxControl,
+	Flex,
+	FlexBlock,
+	FlexItem,
+	Placeholder,
+	Icon,
+} from '@wordpress/components';
+import { useState, useEffect, useCallback } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import './editor.scss';
 import icon from './icon';
 
-function AuthorRecommendationEdit( { className, noticeUI } ) {
+export function AuthorRecommendationEdit( {
+	className,
+	noticeUI,
+	attributes,
+	setAttributes,
+	isSelected,
+} ) {
+	const [ subscriptions ] = useState( [] );
+	const [ selectedSubscriptions, setSelectedSubscriptions ] = useState( [] );
+	const { recommendations } = attributes;
+
+	useEffect( () => {
+		setSelectedSubscriptions( recommendations.map( ( { ID } ) => ID ) );
+		// TODO fetch the sites the user is subscribed to
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	useEffect( () => {
+		setAttributes( {
+			recommendations: subscriptions.filter( subscription =>
+				selectedSubscriptions.includes( subscription.ID )
+			),
+		} );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Do we need to ignore here if we provide `selectedSubscriptions` below? Could we add `setAttributes` and `subscriptions` to the array?